### PR TITLE
Fix: unfold const result with has_null src for string_func_const

### LINKED ISSUE #3554: Query result error in norec mode
### Steps to reproduce the behavior (Required)
1. DROP DATABASE IF EXISTS n1;
CREATE DATABASE n1;
USE n1;
admin set frontend config ("enable_decimal_v3" = "true");
CREATE TABLE IF NOT EXISTS t0 (c_0_0 CHAR(11) NOT NULL ,c_0_1 CHAR(1) NULL ,c_0_2 DATETIME NOT NULL ,c_0_3 DOUBLE NULL ,c_0_4 DATETIME NOT NULL ,c_0_5 BOOLEAN NULL ,c_0_6 BOOLEAN NOT NULL ,c_0_7 DOUBLE NOT NULL ,c_0_8 DECIMAL(34, 4)  NOT NULL ,c_0_9 DATE NOT NULL ,c_0_10 VARCHAR(1) NOT NULL ,c_0_11 VARCHAR(11) NOT NULL ,c_0_12 BOOLEAN NULL ,c_0_13 CHAR(41) NULL ) UNIQUE KEY (c_0_0,c_0_1,c_0_2) DISTRIBUTED BY HASH (c_0_0,c_0_2) properties("replication_num"="3") ;
CREATE TABLE t1 (c_1_0 DECIMAL(15, 8)  NOT NULL ,c_1_1 DATETIME NULL ,c_1_2 DOUBLE NULL ) DUPLICATE KEY (c_1_0,c_1_1) DISTRIBUTED BY HASH (c_1_1) properties("replication_num"="3") ;
INSERT INTO t0 VALUES ("C", "", "1970-01-01 08:00:00", 0.04937511122495242, "1970-01-01 08:00:00", true, false, 6.448814E7, 70403998429948073641798655.9416, "1970-01-01", "", "t", false, "1e");
INSERT INTO t0 VALUES ("", "", "1970-01-01 08:00:00", 0.6589164838554622, "1970-01-01 08:00:00", false, false, 3.33258129E8, 84804939623561703868284178.3980, "1970-01-01", "", "7", false, "WsMThyY8"), ("P", "", "1970-01-01 08:00:00", 0.7862137676291155, "1970-01-01 08:00:00", false, true, 0.43905354561327636, 31769991656533249537593954.7699, "1970-01-01", "", "", true, "4U3"), ("", "", "1970-01-01 08:00:00", NULL, "1970-01-01 08:00:00", false, true, 0.26782181128468363, 69161473128458491705829649.3763, "1970-01-01", "", "", true, "tfb1sLy");
INSERT INTO t1 VALUES (0.01690979, "1970-01-01 08:00:00", 0.5826989699943933), (0.01814800, "1970-01-01 08:00:00", 0.40958542313272484);
INSERT INTO t1 VALUES (0.01448277, NULL, NULL), (0.06258947, NULL, 1.36274051E8);
INSERT INTO t0 VALUES ("IQDIf", "", "1970-01-01 08:00:00", 0.12254909016592053, "1970-01-01 08:00:00", true, false, 0.24859260600176902, 78558395739778525822431420.8774, "1970-01-01", "", "UV", false, "w"), ("8", NULL, "1970-01-01 08:00:00", 0.9122716383558596, "1970-01-01 08:00:00", true, true, 0.6292762544910545, 81308327695679230823048166.9539, "1970-01-01", "", "Km", false, NULL);
INSERT INTO t1 VALUES (0.09740492, "1970-01-01 08:00:00", NULL);
INSERT INTO t0 VALUES ("d", "", "1970-01-01 08:00:00", 2.009140462E9, "1970-01-01 08:00:00", false, false, 0.36926506332423914, 1424326508198447514680795.5781, "1970-01-01", "", "APYtZ", false, "p"), ("", "", "1970-01-01 08:00:00", 0.10122638121677985, "1970-01-01 08:00:00", false, true, 0.9644977109353677, 13832828017466148635069567.1632, "1970-01-01", "", "FPRoHfo", false, "");
INSERT INTO t0 VALUES ("", NULL, "1970-01-01 08:00:00", 0.8909493269742194, "1970-01-01 08:00:00", true, false, 0.7689375909271086, 79083947693114806581488052.1173, "1970-01-01", "", "i", true, ""), ("a", "", "1970-01-01 08:00:00", NULL, "1970-01-01 08:00:00", true, true, 0.31247760448152695, 42922659006024306979775184.7005, "1970-01-01", "", "", false, "QncrLsL"), ("0XWBLGq", "", "1970-01-01 08:00:00", 0.5804378023297032, "1970-01-01 08:00:00", true, true, 0.032645973833674735, 69877245397448621353230224.7841, "1970-01-01", "", "oI", false, "nP");
INSERT INTO t0 VALUES ("", "", "1970-01-01 08:00:00", 0.18411560994229437, "1970-01-01 08:00:00", false, false, 0.004380987639325817, 59254793059282054156804653.2887, "1970-01-01", "", "wM", false, "anoUm4JA"), ("", "", "1970-01-01 08:00:00", 0.24251911422613537, "1970-01-01 08:00:00", false, true, 0.2975838173105556, 56009554615752109339145148.3256, "1970-01-01", "", "6e", NULL, ""), ("", "", "1970-01-01 08:00:00", 0.6509142493548739, "1970-01-01 08:00:00", NULL, true, 0.7399872048435295, 29532409362035368198123874.7056, "1970-01-01", "", "", true, "");
INSERT INTO t0 VALUES ("5I", "", "1970-01-01 08:00:00", 0.6898361589588227, "1970-01-01 08:00:00", NULL, true, 0.2364377190878859, 29652176896635798073277756.6146, "1970-01-01", "", "Qb2akFeQ6", false, "Kk"), ("kB6UzEj", "", "1970-01-01 08:00:00", 0.6890676522335191, "1970-01-01 08:00:00", false, false, 0.3393457467800418, 65097357985470485521489939.2734, "1970-01-01", "", "", NULL, "mN");
INSERT INTO t1 VALUES (0.06548653, "1970-01-01 08:00:00", 0.3619880420177486), (0.05724289, "1970-01-01 08:00:00", 0.9406864918788578);
INSERT INTO t0 VALUES ("l", NULL, "1970-01-01 08:00:00", 0.5232815282486475, "1970-01-01 08:00:00", false, false, 0.3363595223440371, 53313745077934845030144250.7992, "1970-01-01", "", "B", false, ""), ("bIm", "", "1970-01-01 08:00:00", -1.397810988E9, "1970-01-01 08:00:00", false, true, 0.8527495091215096, 73102537697689303457159182.2867, "1970-01-01", "", "10", NULL, "CUAQ5"), ("", "", "1970-01-01 08:00:00", 0.35764266865366723, "1970-01-01 08:00:00", false, true, 0.12779015586363618, 97219895094052688254676608.5368, "1970-01-01", "", "", true, NULL);

2. mysql> SELECT COUNT(*) FROM  (SELECT t1.c_1_0, t1.c_1_1, t1.c_1_2 FROM t1 WHERE  ( t1.c_1_1 ) > ( CASE  WHEN CASE  WHEN false THEN (t1.c_1_0) IN (NULL, 0.07038423)   WHEN true THEN true  WHEN  ( t1.c_1_1 ) >= ( t1.c_1_1 )  THEN  ( t1.c_1_0 ) != ( t1.c_1_0 )  END THEN t1.c_1_1  WHEN (NOT (((t1.c_1_0) IS NOT NULL))) THEN t1.c_1_1 END ) ) subt1 RIGHT OUTER JOIN t0 ON subt1.c_1_2 = t0.c_0_7 AND subt1.c_1_2 < t0.c_0_7 AND subt1.c_1_1 <= t0.c_0_9 AND subt1.c_1_1 >= t0.c_0_9 AND subt1.c_1_2 > t0.c_0_9 AND subt1.c_1_2 != t0.c_0_9 WHERE ((((RPAD(CAST(t0.c_0_3 AS STRING ) , YEARS_DIFF("1969-12-20 17:33:12", "1970-01-02 00:04:07") , REGEXP_EXTRACT("", "", 949246558) ) )||((("r")||("2"))))) IS NOT NULL);
+----------+
| count(*) |
+----------+
|       13 |
+----------+
1 row in set (0.03 sec)

mysql> SELECT SUM(count) FROM (SELECT CAST((CAST(((((RPAD(CAST(t0.c_0_3 AS STRING ) , YEARS_DIFF("1969-12-20 17:33:12", "1970-01-02 00:04:07") , REGEXP_EXTRACT("", "", 949246558) ) )||((("r")||("2"))))) IS NOT NULL) AS BOOLEAN) = true) AND (CAST(((((RPAD(CAST(t0.c_0_3 AS STRING ) , YEARS_DIFF("1969-12-20 17:33:12", "1970-01-02 00:04:07") , REGEXP_EXTRACT("", "", 949246558) ) )||((("r")||("2"))))) IS NOT NULL) AS BOOLEAN) IS NOT NULL) AS INT) as count FROM   (SELECT t1.c_1_0, t1.c_1_1, t1.c_1_2 FROM t1 WHERE  ( t1.c_1_1 ) > ( CASE  WHEN CASE  WHEN false THEN (t1.c_1_0) IN (NULL, 0.07038423)   WHEN true THEN true  WHEN  ( t1.c_1_1 ) >= ( t1.c_1_1 )  THEN  ( t1.c_1_0 ) != ( t1.c_1_0 )  END THEN t1.c_1_1  WHEN (NOT (((t1.c_1_0) IS NOT NULL))) THEN t1.c_1_1 END ) ) subt1 RIGHT OUTER JOIN t0 ON subt1.c_1_2 = t0.c_0_7 AND subt1.c_1_2 < t0.c_0_7 AND subt1.c_1_1 <= t0.c_0_9 AND subt1.c_1_1 >= t0.c_0_9 AND subt1.c_1_2 > t0.c_0_9 AND subt1.c_1_2 != t0.c_0_9) t;
+--------------+
| sum(`count`) |
+--------------+
|           12 |
+--------------+

### StarRocks version (Required)
 - You can get the StarRocks version by executing SQL `select current_version()`
 - branch-2.1 Release
 - mysql> select current_version();
+---------------------------------+
| current_version()               |
+---------------------------------+
| SQLANCER_01_RELEASE 98431ea |
+---------------------------------+

### PR DESCRIPTION
**Pull request by ZiheLiu**
## What type of PR is this：
- [x] bug
- [ ] feature
- [ ] enhancement
- [ ] others

## Which issues of this PR fixes ：

Fixes #3554.

## Problem Summary：

In `string_func_const`, if the result is constant column, it will return it directly.
However, `is_null` of all the rows may be different. For example, `rpad(['abc', null], 0, '')` should return `['', null]`, instead of `['', '']`.

Therefore,  we should unfold the constant result column, if some rows are null.

### CODE REVIEW

File: be/src/exprs/vectorized/string_functions.cpp [hunk 1]
```diff
@@ -539,11 +539,17 @@ ColumnPtr string_func_const(StringConstFuncType func, const Columns& columns, Ar
...
+            // - if binary is non-null ConstColumn, unfold it and wrap with src_null.
+            // - if binary is NullableColumn, then the null column inside the binary should
+            //   be merged with the null column inside of columns[0].
             if (binary->is_constant()) {
-                return binary;
+                if (binary->only_null()) {
+                    return binary;
+                }
+                ConstColumn* dst_const = down_cast<ConstColumn*>(binary.get());
+                dst_const->data_column()->assign(dst_const->size(), 0);
```
Comment (satanson) [inline be/src/exprs/vectorized/string_functions.cpp:551]: only_null returning true indicates that the column is const null.
so break the logic into cases as follows:
1. const null:  only_null()
2. constant column: is_constant()
3. nullable column: is_nullable()
4. elementry column: else

Comment (ZiheLiu) [reply be/src/exprs/vectorized/string_functions.cpp:551]: Fixed.

**Comment by ZiheLiu** [comment]
run starrocks_be_unittest

File: be/test/exprs/vectorized/string_fn_pad_test.cpp [hunk 2]
```diff
@@ -511,5 +511,164 @@ TEST_F(StringFunctionPadTest, rpadConstTest) {
...
+        if (!c.lpad_expected_nulls[i]) {
+            ASSERT_EQ(c.lpad_expected_results[i], lpad_v->get_data()[i].to_string()) << "Row#" << i;
+        }
+    }
+}
+
+TEST_F(StringFunctionPadTest, padNullableStrConstLenFillTest) {
+    std::vector<PadNullableStrConstLenFillTestCase> cases = {
+            {// Input str, len, fill.
+             {"<NULL>", "<NULL>"},
```
[the remark this answers is not in the document]
Comment (ZiheLiu) [reply be/test/exprs/vectorized/string_fn_pad_test.cpp]: `strs[i]`, `str_nulls[i]`, `rpad_expected_results[i]` etc. are the i-th input `str` and result.

**Review by Pslydhh — APPROVED** [review]

**Review by kangkaisen — APPROVED** [review]

**Review by stdpain — APPROVED** [review]

**Pull request merged**